File: src/test/java/no/nav/permitteringsskjemaapi/PermitteringsskjemaRepositoryTest.java
package no.nav.permitteringsskjemaapi;

import no.nav.permitteringsskjemaapi.permittering.Permitteringsskjema;
import no.nav.permitteringsskjemaapi.permittering.PermitteringsskjemaJuridiskEnhet;
import no.nav.permitteringsskjemaapi.permittering.PermitteringsskjemaJuridiskEnhetRepository;
import no.nav.permitteringsskjemaapi.permittering.PermitteringsskjemaRepository;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.context.annotation.Profile;
import org.springframework.test.annotation.DirtiesContext;
import org.springframework.test.context.ActiveProfiles;
import org.springframework.test.context.junit4.SpringRunner;

import java.util.List;
import java.util.Optional;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(SpringRunner.class)
@SpringBootTest
@Profile("test")
@DirtiesContext
public class PermitteringsskjemaRepositoryTest {
    private Permitteringsskjema lagretPermitteringsskjema;
    private Permitteringsskjema permitteringsskjema;

    private PermitteringsskjemaJuridiskEnhet permitteringsskjemaJuridiskEnhet;
    private PermitteringsskjemaJuridiskEnhet lagretPermitteringsskjemaJuridiskEnhet;

    @Autowired
    private PermitteringsskjemaRepository repository;
    @Autowired
    private PermitteringsskjemaJuridiskEnhetRepository juridiskEnhetRepository;

    @Before
    public void setUp() {
        permitteringsskjema = PermitteringTestData.enPermitteringMedAltFyltUt();
        permitteringsskjemaJuridiskEnhet = PermitteringTestData.enJuridiskEnhetUtenSkjemaer();

        lagretPermitteringsskjema = repository.save(permitteringsskjema);
        lagretPermitteringsskjemaJuridiskEnhet = juridiskEnhetRepository.save(permitteringsskjemaJuridiskEnhet);
    }

    @Test
    public void skal_kunne_lagre_alle_felter() {
        assertThat(lagretPermitteringsskjema)
                .usingRecursiveComparison() // Felt-for-felt sammenligning
                .isEqualTo(permitteringsskjema);
        assertThat(lagretPermitteringsskjemaJuridiskEnhet)
                .usingRecursiveComparison()
                .isEqualTo(permitteringsskjemaJuridiskEnhet);
    }

    @Test
    public void skal_lagre_og_hente_juridisk_enhet_med_liste_av_underenheter() {
        PermitteringsskjemaJuridiskEnhet permitteringsskjemaJuridiskSkalHaEnOrganisasjon = PermitteringTestData.enJuridiskEnhetUtenSkjemaer();
        Permitteringsskjema permitteringsskjemaOrganisasjon = PermitteringTestData.enPermitteringMedAltFyltUt();
        permitteringsskjemaOrganisasjon.setPermitteringsskjemaJuridiskEnhet(permitteringsskjemaJuridiskSkalHaEnOrganisasjon);
        permitteringsskjemaJuridiskSkalHaEnOrganisasjon.setPermitteringsskjemaer(List.of(permitteringsskjemaOrganisasjon));
        PermitteringsskjemaJuridiskEnhet juridiskEnhetMedListeAvOrganisasjoner = juridiskEnhetRepository.save(permitteringsskjemaJuridiskSkalHaEnOrganisasjon);

        assertThat(juridiskEnhetMedListeAvOrganisasjoner.getPermitteringsskjemaer()).isNotEmpty();
    }

    @Test
    public void skal_kunne_hentes_med_id() {
        Optional<Permitteringsskjema> hentetPermittering = repository.findById(permitteringsskjema.getId());
        assertThat(hentetPermittering).hasValue(permitteringsskjema);
    }
}

File: src/test/kotlin/no/nav/permitteringsskjemaapi/PermitteringsskjemaRepositoryTest.kt
package no.nav.permitteringsskjemaapi

import no.nav.permitteringsskjemaapi.permittering.PermitteringsskjemaRepository
import no.nav.permitteringsskjemaapi.permittering.testSkjema
import org.flywaydb.core.Flyway
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.ActiveProfiles
import java.util.*


@SpringBootTest
@ActiveProfiles("test")
class PermitteringsskjemaRepositoryTest {
    @Autowired
    lateinit var repository: PermitteringsskjemaRepository

    @Autowired
    lateinit var flyway: Flyway

    @BeforeEach
    fun clearDatabase() {
        flyway.clean()
        flyway.migrate()
    }

    @Test
    fun skal_kunne_lagre_alle_felter() {
        val permitteringsskjema = testSkjema(bedriftNr = "999999")
        repository.save(permitteringsskjema)
    }

    @Test
    fun skal_kunne_hentes_med_id() {
        val id = UUID.randomUUID()
        val permitteringsskjema = testSkjema(id = id, bedriftNr = "999999")
        repository.save(permitteringsskjema)
        val hentetPermittering = repository.findById(id)

        assert(hentetPermittering != null)
    }
}